src/legacy/core_plugins/kibana/public/dashboard/dashboard_empty_screen.tsx

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { I18nProvider, FormattedMessage } from '@kbn/i18n/react';
+import { EuiIcon, EuiLink } from '@elastic/eui';
+import * as constants from './dashboard_empty_screen_constants';
+
+export interface Props {
+  showLinkToVisualize: boolean;
+  onLinkClick: () => void;
+}
+
+export function DashboardEmptyScreen({ showLinkToVisualize, onLinkClick }: Props) {
+  const linkToVisualizeParagraph = (
+    <p className="linkToVisualizeParagraph">
+      <FormattedMessage
+        id="kbn.dashboard.addVisualizationDescription3"
+        defaultMessage="If you haven't set up any visualizations yet, {visualizeAppLink} to create your first visualization"
+        values={{
+          visualizeAppLink: (
+            <a className="euiLink" href="#/visualize">
+              {constants.visualizeAppLinkTest}
+            </a>
+          ),
+        }}
+      />
+    </p>
+  );
+  const paragraph = (
+    description1: string,
+    description2: string,
+    linkText: string,
+    ariaLabel: string,
+    dataTestSubj?: string
+  ) => {
+    return (
+      <p>
+        <span>
+          {description1}
+          <EuiLink onClick={onLinkClick} aria-label={ariaLabel} data-test-subj={dataTestSubj || ''}>
+            {linkText}
+          </EuiLink>
+          {description2}
+        </span>
```

Comment:
Why not implementing it like in `linkToVisualizeParagraph`? Because in another language you might have a different order of the parts of the sentence, and it's much harder to get the context for the translator this way.

Using `Click the {buttonLinkName} button in the menu bar above to add a visualization to the dashboard.` with replacement would give context, an also allow easy translations for Jedis like `{buttonLinkName} button you must click ...`

Reply:
I agree it is a great i18n anti-pattern. However, this is a literal c/p from Angular code. This screen is to change soon, so all of these strings will be gone and replaced with something nicer:
<img width="668" alt="Screenshot 2019-11-21 at 10 53 03" src="https://user-images.githubusercontent.com/1937956/69331762-2d491200-0c4d-11ea-8258-407711d175fc.png">

That's why I decided not to spend too much time on refactoring the existing strings.